gluten-core/src/main/scala/io/glutenproject/execution/WholeStageTransformer.scala

```diff
@@ -96,9 +96,16 @@ trait UnaryTransformSupport extends TransformSupport with UnaryExecNode {
 
 case class WholeStageTransformer(child: SparkPlan, materializeInput: Boolean = false)(
     val transformStageId: Int
-) extends UnaryTransformSupport {
+) extends GenerateTreeStringShim
+  with UnaryTransformSupport {
   assert(child.isInstanceOf[TransformSupport])
 
+  def stageId: Int = transformStageId
+
+  def substraitPlanJsonValue: String = substraitPlanJson
```

Comment:
Can we remove the new method `substraitPlanJsonValue` and just use the existing `substraitPlanJson`?  We may just need to declare `substraitPlanJson` in the extending trait.

Reply:
Sure. I will updated in the following PRs. Thanks @PHILO-HE